src/main/java/org/example/GrandPiano.java

```diff
@@ -0,0 +1,20 @@
+//このクラスは、org.exampleパッケージに属す
+package org.example;
+
+//パブリック設定でGrandPianoという名前のクラスを作成、Pianoという名前のクラスを継承せよ
+public class GrandPiano extends Piano {
+
+  //パブリック設定でString型の引数instrumentNameを受け取るplayという名前のメソッドを作成せよ
+  public String play(String instrumentName) {
+
+    //引数instrumentNameと"素晴らしい!"を連結した文字列を呼び出し元に返せ
+    return super.play(instrumentName) + "素晴らしい!";
+  }

```

Comment:
グランドピアノであるということを引数に取る必要はないのではないでしょうか？
```suggestion
  //パブリック設定でString型の引数instrumentNameを受け取るplayという名前のメソッドを作成せよ
  public String play() {

    //引数instrumentNameと"素晴らしい!"を連結した文字列を呼び出し元に返せ
    return this. instrumentName + "素晴らしい!";
  }

```

Reply:
```java
public class GrandPiano extends Piano {
  public static final String INSTRUMENT_NAME = "グランドピアノ";
```
グランドピアノかどうかというのはクラスに属するものなのでクラス変数でいいかなと思いました。

Reply:
public String play() {
    return INSTRUMENT_NAME + "を演奏しています! 素晴らしい!";
}

Reply:
> グランドピアノであるということを引数に取る必要はないのではないでしょうか？
GrandPiano クラスは既に Piano クラスを継承しているので、instrumentName として「グランドピアノ」を指定する必要がないという認識であっていますか？

Reply:
> ```java
> public class GrandPiano extends Piano {
>   public static final String INSTRUMENT_NAME = "グランドピアノ";
> ```
> 
> グランドピアノかどうかというのはクラスに属するものなのでクラス変数でいいかなと思いました。

調べたところ、

Reply:
> グランドピアノであるということを引数に取る必要はないのではないでしょうか？

**◆【質問したいこと】**
1. GrandPianoクラスのplayメソッドが引数なしの場合、GrandPianoクラスのplayメソッドで付け加えた、"素晴らしい!"という文字が出力されません。
引数なしでよいのでしょうか？

2. GrandPianoクラスで正しくオーバーライドするためには、引数を受け取るようにしなくてはならないのでは、と思ったのですが、合っていますか？
オーバーライドしなくてもよいのであれば、ご指摘の通り、引数なしでOKなのかな、と思ったのですが…
(認識がずれていたら、すみません)


3. 以下のコードへの変更が前提の場合について、教えてください
**◆【質問したいこと】**
1. GrandPianoクラスのplayメソッドが引数なしの場合、GrandPianoクラスのplayメソッドで付け加えた、"素晴らしい!"という文字が出力されません。
引数なしでよいのでしょうか？

2. GrandPianoクラスで正しくオーバーライドするためには、引数を受け取るようにしなくてはならないのでは、と思ったのですが、合っていますか？
オーバーライドしなくてもよいのであれば、ご指摘の通り、引数なしでOKなのかな、と思ったのですが…
(認識がずれていたら、すみません)


3. 以下のコードへの変更が前提の場合について、教えてください。

```
public class GrandPiano extends Piano {
  public static final String INSTRUMENT_NAME = "グランドピアノ";
```

もし引数を受け取るようにしないといけないのであれば、以下のどちらかのコードに変更すればよいという認識で合っていますか？
どちらのコードの場合でも、出力結果は「グランドピアノを演奏しています!素晴らしい!」です。


```
public String play(String instrumentName) {
    return INSTRUMENT_NAME + "を演奏しています!素晴らしい！";
    }
```

```
public String play(String instrumentName) {
   return super.play(instrumentName) + "素晴らしい!";
 }
```

【調べたこと】
GrandPianoクラスのplayメソッドが引数なしで定義されているため、Pianoクラスのplayメソッドがオーバーライドされていないようです。
(GrandPianoクラスのplayメソッドが引数を受け取らないため、Pianoクラスのplayメソッドが呼ばれているようです。)。

```
public class GrandPiano extends Piano {
  public static final String INSTRUMENT_NAME = "グランドピアノ";
```

もし引数を受け取るようにしないといけないのであれば、以下のどちらかのコードに変更すればよいという認識で合っていますか？
どちらのコードの場合でも、出力結果は「グランドピアノを演奏しています!素晴らしい!」です。


```
public String play(String instrumentName) {
    return INSTRUMENT_NAME + "を演奏しています!素晴らしい！";
    }
```

```
public String play(String instrumentName) {
   return super.play(instrumentName) + "素晴らしい!";
 }
```

【調べたこと】
GrandPianoクラスのplayメソッドが引数なしで定義されているため、Pianoクラスのplayメソッドがオーバーライドされていないようです。
(GrandPianoクラスのplayメソッドが引数を受け取らないため、Pianoクラスのplayメソッドが呼ばれているようです。)

Reply:
> グランドピアノかどうかというのはクラスに属するものなのでクラス変数でいいかなと思いました。

> ```java
> public class GrandPiano extends Piano {
>   public static final String INSTRUMENT_NAME = "グランドピアノ";
> ```
> 
> グランドピアノかどうかというのはクラスに属するものなのでクラス変数でいいかなと思いました。

以下、調べたことをまとめました。

【調べたこと】
・GrandPiano クラスが「グランドピアノ」であることはそのクラスに固有の属性(フィールド)であるので、クラス変数 (static final) として定義するのが適切。
クラス変数を使用することで、コードが一貫性を持ち、明確になる。

・final を付けると、その変数の値は変更できなくなる。
変数やフィールドの値を変更不可能にしたり、メソッドのオーバーライドを禁止したり、クラスの継承を禁止するのに使われる。




 ```
public String play() {
    return INSTRUMENT_NAME + "を演奏しています! 素晴らしい!";
}
 ```


 ```
public class GrandPiano extends Piano {
  public static final String INSTRUMENT_NAME = "グランドピアノ";
    public String play() {
        return INSTRUMENT_NAME;
    }
 ```
→　グランドピアノを演奏しています!

Reply:
> グランドピアノかどうかというのはクラスに属するものなのでクラス変数でいいかなと思いました。

クラス変数に修正しました！